lib/xdrgen/generators/go.rb

```diff
@@ -400,17 +400,17 @@ def render_enum_encode_to_interface(out, typedef)
         EOS
       end
 
-      def is_fixed_opaque(type)
-        type.is_a?(AST::Typespecs::Opaque) && type.fixed?
+      def results_in_go_array(type)
+        (type.is_a?(AST::Typespecs::Opaque) && type.fixed?) || type.sub_type == :array
       end
 
       def render_typedef_encode_to_interface(out, typedef)
         name = name(typedef)
         type = typedef.declaration.type
         out.puts "// EncodeTo encodes this value using the Encoder."
-        if is_fixed_opaque(type) ||
-           (type.is_a?(AST::Identifier) && type.resolved_type.is_a?(AST::Definitions::Typedef) && is_fixed_opaque(type.resolved_type.declaration.type))
-          # Implement EncodeTo by pointer in fixed opaque types (arrays)
+        if results_in_go_array(type) ||
+            (type.is_a?(AST::Identifier) && type.sub_type == :simple && type.resolved_type.is_a?(AST::Definitions::Typedef) && results_in_go_array(type.resolved_type.declaration.type))
```

Comment:
It checks that it's a type synonym (from identifier to identifier), or at least that was my intention.

Reply:
Do you need the condition? I'm surprised it's required, because if a resolved type is typedef and the resolved type a fixed length array then it feels appropriate to do this too.

Reply:
I do, otherwise the code generates EncodeTo by pointer for array types (as indicated in the commit message)

Reply:
Ahh, yes, makes sense.